src/redux/features/authDetailsSlice.ts

```diff
@@ -0,0 +1,32 @@
+import { Token } from '@/types/common'
+import { createSlice } from '@reduxjs/toolkit'
+import { RootState } from '@/redux/store'
+
+interface IInitialState {
+  tokenPayload: Token | {}
```

Comment:
Instead of `{}`, let's type this as `undefined` for undefined cases. The return type of `getTokenPayload()` method is `Token | null`, so you might want to consider this as well. 
`undefined` type is more suitable than `{}` for the following reasons:
- Semantic Clarity: By using undefined, you explicitly communicate that the tokenPayload property may not have a value initially. This aligns better with the intent of the code and makes it clearer for other developers who might work with the code in the future.
- Type Safety: Using undefined allows TypeScript to enforce stricter type checks. If you attempt to access properties or methods on tokenPayload without first checking if it's defined, TypeScript will warn you, helping to catch potential bugs at compile-time.
- Avoids Ambiguity: When you use {}, it could imply an empty object ({}), which might lead to confusion regarding whether the property is expected to be an object with properties or just an empty placeholder.